diff --git a/tools/update-components.sh b/tools/update-components.sh
index bd0e394..e9724cb 100755
--- a/tools/update-components.sh
+++ b/tools/update-components.sh
@@ -80,10 +80,11 @@ fi
 echo "Updating ESP-LITTLEFS..."
 if [ ! -d "$AR_COMPS/esp_littlefs" ]; then
 	git clone $LITTLEFS_REPO_URL "$AR_COMPS/esp_littlefs" && \
+	git -C "$AR_COMPS/esp_littlefs" checkout --detach v1.16.4 && \
     git -C "$AR_COMPS/esp_littlefs" submodule update --init --recursive
 else
 	git -C "$AR_COMPS/esp_littlefs" fetch && \
-	git -C "$AR_COMPS/esp_littlefs" pull --ff-only && \
+	git -C "$AR_COMPS/esp_littlefs" checkout --detach v1.16.4 && \
     git -C "$AR_COMPS/esp_littlefs" submodule update --init --recursive
 fi
 if [ $? -ne 0 ]; then exit 1; fi
